src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -47,6 +50,19 @@ private HttpSender getHttpSender() {
 
 	@Override
 	protected Void doInBackground() throws Exception {
+		TreeSet<HtmlParameter> cookies = this.httpMessage.getRequestHeader().getCookieParams();
+		if (targetToken.getType() == HtmlParameter.Type.cookie) {
+			Iterator<HtmlParameter> iter = cookies.iterator();
+			while (iter.hasNext()) {
+				HtmlParameter cookie = iter.next();
+				if (cookie.getName().equals(targetToken.getName())) {
+					iter.remove();
+					break;
+				}
+			}
+		}
+		HttpMessage msg = this.httpMessage.cloneRequest();
+		msg.getRequestHeader().setCookieParams(cookies);

```

Comment:
Only the cookie that you want to sequence should be removed. It might happen that you want to sequence something after you authenticated, then you want that authentication cookie to be present, i.e. a CSRF token after a login.